components/higher-order/with-state/test/index.js

```diff
@@ -1,13 +1,29 @@
 /**
  * External dependencies
  */
-import { mount } from 'enzyme';
+import TestUtils from 'react-dom/test-utils';
 
 /**
  * Internal dependencies
  */
 import withState from '../';
 
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+
+// this is needed because TestUtils does not accept a stateless component.
+// anything run through a HOC ends up as a stateless component.
+const getTestComponent = ( WrappedComponent ) => {
```

Comment:
We are using a similar HOC in other places should we implement it in a place where we can reuse it?

Reply:
:)  Currently I've got pulls open for all the affected HOCs that implement this (or a variation of) when needed.  I think there's definitely room for abstraction but I think that should be part of a larger discussion (via its own pull/issue) about creating our own abstraction for testing components.  I'd like to focus on getting these pulls merged in first and then we can accumulate patterns that we notice across the various tests as a part of the discussion on the abstraction (I can contribute to that with patterns I noticed when doing these conversions).

Reply:
Also the current comment is _currently_ against master branch incorrect.  When/if #7557 gets implemented, then it will be completely correct because every HOC (created by `createHigherOrderComponent` will be wrapped by forwardRef).

Reply:
> We are using a similar HOC in other places should we implement it in a place where we can reuse it?

We definitely should do it once we merge all PRs :) I'd prefer to do it in one go so we large enough codebase to sparkle proper discussion.